### ISSUE TITLE
Generating Tags runs in Parallel despite concurrency: 1 - causes issues when using jib with concurrent mavens

### ISSUE THREAD

**Issue by ham1** [GoogleContainerTools/skaffold#7071]
### Expected behaviour

When using jib for multiple images, generating tags should not run more than 1 maven command at the same time.

At the very least the concurrency of this stage should be configurable either by matching `concurrency` or via another property.

This also causes excessive load if using `-T 1C` is used in the jib/maven config.

### Actual behaviour

All images run this in parallel:
```
 /usr/lib/jvm/java-11-openjdk-amd64//bin/java -classpath /[...]/apache-maven-3.8.4/boot/plexus-classworlds-2.6.0.jar -Dclassworlds.conf=/[...]/apache-maven-3.8.4/bin/m2.conf -Dmaven.home=/[...]/apache-maven-3.8.4 -Dlibrary.jansi.path=/[...]/apache-maven-3.8.4/lib/jansi-native -Dmaven.multiModuleProjectDirectory=/[...]/src org.codehaus.plexus.classworlds.launcher.Launcher jib:_skaffold-fail-if-jib-out-of-date -Djib.requiredVersion=1.4.0 -Dmaven.test.skip=true --threads=1C --strict-checksums --projects java/[...] --also-make jib:_skaffold-files-v2 --quiet --batch-mode
```

Thus frequently causing issues with 0 byte jars and `Malformed \\uxxxx encoding.` in the resolver-status.properties.

### Information

- Skaffold version: v1.35.2
- Operating system: Ubuntu Linux 20.04.3 LTS
- Installed via: `curl -Lo skaffold https://storage.googleapis.com/skaffold/releases/v1.35.2/skaffold-linux-amd64`
- Contents of skaffold.yaml:

```yaml
<paste your skaffold.yaml here>
apiVersion: skaffold/v2beta14
kind: Config

build:
  local:
    useBuildkit: true
    concurrency: 1
    tryImportMissing: true
  tagPolicy:
    inputDigest: {}
  artifacts:
    - image: [...]
      context: src
      jib:
        project: java/[...]
        args: [ '--threads=1C','--strict-checksums' ]
        fromImage: [...]/java-8-base
      requires:
        - image: [...]/java-8-base
    - image: [...]
      context: src
      jib:
        project: java/[...]
        args: [ '-Dmaven.test.skip=true','--threads=1C','--strict-checksums' ]
        fromImage: [...]/java-8-base
      requires:
      - image: [...]/java-8-base
      [...]
```

### Steps to reproduce the behaviour

1. Anything with more than 1 image that uses jib.
2. skaffold build
3. (we also have some Bazel targets)
```
skaffold build                                                                                                                                                       
Generating tags...                                                                                                                                                                                                 
 - [...]-> WARN[0001] Retrieving Bazel dependencies can take a long time the first time  subtask=-1 task=Build
localhost:32000/[...]
[...]
Checking cache...
 - [...]: Not found. Building
 - [...]: Found Remotely
 -[...]: Not found. Building
[...]
```

**Comment by tejal29** [comment]
@ham1 Can you attach the debug logs for the latest version. We recently fixed #7181

Please re-open if the latest does not work

**Comment by ham1** [comment]
Ok, thank you for replying!

After looking at the code I've managed to make `v1.37.0` "work" for us (i.e. generate one tag at a time) by setting `GOMAXPROCS=1`. e.g. `GOMAXPROCS=1 skaffold build` but this doesn't feel optimal and isn't documented/intended/stable?

https://github.com/GoogleContainerTools/skaffold/blob/1a431da621b45649473dcb9738680c6f3f03a0c4/pkg/skaffold/runner/build.go#L163

Could we change this to use the concurrency configuration, as I feel that would be more intuitive?
